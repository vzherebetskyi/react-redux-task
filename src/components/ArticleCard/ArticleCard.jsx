import React from 'react';
import moment from 'moment';

import './ArticleCard.scss';
import ImagePlaceholder from '../../assets/images/placeholder-image.webp';

const ArticleCard = ({ article }) => (
    <div className='card-wrapper' onClick={() => window.open(article.url,'_blank')}>
      <p className='card-title'>{article.title}</p>
      <div className='card-image-container'>
        <img src={article.urlToImage ? article.urlToImage : ImagePlaceholder} className='card-image' />
      </div>
      <div className='card-info-container'>
        <div className='card-info-container__author'>
          <p><span>Author:</span> {article.author}</p>
        </div>
        <div className='card-info-container__published'>
          <p><span>Published:</span> {moment(article.publishedAt).format('Do MMMM YYYY')}</p>
        </div>
        <div className='card-info-container__source'>
          <p><span>Source:</span> {article.source}</p>
        </div>
      </div>
      <div className='card-description-container'>
        <p className='card-description-text'>{article.content}</p>
        <p><strong>API:</strong> {article.api}</p>
      </div>
    </div>
  );

export default ArticleCard;
