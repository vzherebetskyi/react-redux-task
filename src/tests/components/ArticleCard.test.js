import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import ArticleCard from '../../components/ArticleCard/ArticleCard';

test('should render ArticleCard component correctly', () => {
  const elem = renderer
    .create(
      <ArticleCard
        article={{
          author: 'No info',
          content: 'No info',
          publishedAt: '2024-08-18T07:26:36Z',
          unifiedDate: '18th August 2024',
          source: 'No info',
          title: 'No info',
          url: null,
          urlToImage: null,
          category: 'mixed',
          api: 'test',
        }}
      />,
    )
    .toJSON();
  expect(elem).toMatchSnapshot();
});

test('It should render fields of the cards correctly', () => {
  const { container } = render(
    <ArticleCard
      article={{
        author: 'John Doe',
        content: 'Medicine article',
        publishedAt: '2024-08-18T07:26:36Z',
        unifiedDate: '18th August 2024',
        source: 'No info',
        title: 'No info',
        url: null,
        urlToImage: null,
        category: 'mixed',
        api: 'test',
      }}
    />,
  );
  expect(container).toBeTruthy();
  const authorName = screen.getByText('John Doe');
  expect(authorName).toBeInTheDocument();
  const articleContent = screen.getByText('Medicine article');
  expect(articleContent).toBeInTheDocument();
  const articleDate = screen.getByText('18th August 2024');
  expect(articleDate).toBeInTheDocument();
  const articleSourceAndTitle = screen.getAllByText('No info');
  expect(articleSourceAndTitle).toHaveLength(2);
  const articleApi = screen.getByText('test');
  expect(articleApi).toBeInTheDocument();
});
