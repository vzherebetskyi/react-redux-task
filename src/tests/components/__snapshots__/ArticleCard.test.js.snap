// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ArticleCard component correctly 1`] = `
<div
  className="card-wrapper"
  onClick={[Function]}
>
  <p
    className="card-title"
  >
    No info
  </p>
  <div
    className="card-image-container"
  >
    <img
      className="card-image"
      src={
        {
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
    />
  </div>
  <div
    className="card-info-container"
  >
    <div
      className="card-info-container__author"
    >
      <p>
        <span>
          Author:
        </span>
         
        No info
      </p>
    </div>
    <div
      className="card-info-container__published"
    >
      <p>
        <span>
          Published:
        </span>
         
        18th August 2024
      </p>
    </div>
    <div
      className="card-info-container__source"
    >
      <p>
        <span>
          Source:
        </span>
         
        No info
      </p>
    </div>
  </div>
  <div
    className="card-description-container"
  >
    <p
      className="card-description-text"
    >
      No info
    </p>
    <p>
      <strong>
        API:
      </strong>
       
      test
    </p>
  </div>
</div>
`;
